app/multitenant/consul_client.go

```diff
@@ -133,21 +133,40 @@ func (c *consulClient) CAS(key string, out interface{}, f CASCallback) error {
 }
 
 func (c *consulClient) WatchPrefix(prefix string, out interface{}, done chan struct{}, f func(string, interface{}) bool) {
-	index := uint64(0)
+	const (
+		initialBackoff = 1 * time.Second
+		maxBackoff     = 1 * time.Minute
+	)
+	var (
+		backoff = initialBackoff / 2
+		index   = uint64(0)
+	)
 	for {
 		select {
 		case <-done:
 			return
 		default:
 		}
-
 		kvps, meta, err := c.kv.List(prefix, &consul.QueryOptions{
```

Comment:
I was looking for a way to cancel the request directly but there doesn't seem to be a way, meh